File: Spring-Security-DB-Authentication/src/main/java/com/github/spdeepak/security/CustomUserDetails.java
package com.github.spdeepak.security;

import com.github.spdeepak.entity.User;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.stream.Collectors;

public class CustomUserDetails extends User implements UserDetails {

	private static final Logger log = LoggerFactory.getLogger(CustomUserDetails.class);

	public CustomUserDetails(final User user) {
		super(user);
	}

	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		return getRoles()
				.stream()
				.map(role -> {
					log.debug("Granting Authority to user with role: " + role.toString());
					return new SimpleGrantedAuthority(role.toString());
				})
				.collect(Collectors.toList());
	}

	@Override
	public String getPassword() {
		return super.getPassword();
	}

	@Override
	public String getUsername() {
		return super.getUsername();
	}

	@Override
	public boolean isAccountNonExpired() {
		return super.isAccountNonExpired();
	}

	@Override
	public boolean isAccountNonLocked() {
		return super.isAccountNonLocked();
	}

	@Override
	public boolean isCredentialsNonExpired() {
		return super.isCredentialsNonExpired();
	}

	@Override
	public boolean isEnabled() {
		return super.isEnabled();
	}
}

File: Kotlin-Spring-Security-DB-Authentication/src/main/kotlin/com/github/spdeepak/security/CustomUserDetails.kt
package com.github.spdeepak.security

import com.github.spdeepak.entity.User
import org.slf4j.LoggerFactory
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.stream.Collectors

/**
 * @author Deepak Sunanda Prabhakar
 */
open class CustomUserDetails : User, UserDetails {

	private val log = LoggerFactory.getLogger(CustomUserDetails::class.java)

	constructor(user: User) : super(user)


	override fun getAuthorities(): Collection<GrantedAuthority> {
		return roles
				.stream()
				.map { role ->
					log.debug("Granting Authority to user with role: " + role.toString())
					SimpleGrantedAuthority(role.toString())
				}
				.collect(Collectors.toList())
	}

	override fun getPassword(): String {
		return super.passWord
	}

	override fun getUsername(): String {
		return super.userName
	}

	override fun isEnabled(): Boolean {
		return super.enabled
	}

	override fun isCredentialsNonExpired(): Boolean {
		return super.credentialsNonExpired
	}

	override fun isAccountNonExpired(): Boolean {
		return super.accountNonExpired
	}

	override fun isAccountNonLocked(): Boolean {
		return super.accountNonLocked
	}

}
